server/service/src/sync/test/test_data/goods_received_line.rs

```diff
@@ -0,0 +1,118 @@
+// use crate::sync::{
+//     test::TestSyncIncomingRecord, translations::purchase_order_line::LegacyPurchaseOrderLineRow,
+// };
+// use chrono::NaiveDate;
+// use repository::PurchaseOrderLineRow;
+// use serde_json::json;
+
+// use super::TestSyncOutgoingRecord;
+
+use chrono::NaiveDate;
+use repository::{GoodsReceivedLineRow, GoodsReceivedLineRowDelete};
+use serde_json::json;
+
+use crate::sync::{
+    test::{TestSyncIncomingRecord, TestSyncOutgoingRecord},
+    translations::goods_received_line::LegacyGoodsReceivedLineRow,
+};
+
+const TABLE_NAME: &str = "goods_received_line";
+
+const GOODS_RECEIVED_LINE: (&str, &str) = (
+    "917BA6AEC9984FF09F6DB8599CA426B0",
+    r#"{
+        "ID": "917BA6AEC9984FF09F6DB8599CA426B0",
+        "authorised_comment": "",
+        "batch_received": "sal_bat_one",
+        "comment": "",
+        "cost_price": 0,
+        "custom_stock_field_1": "",
+        "custom_stock_field_2": "",
+        "custom_stock_field_3": "",
+        "custom_stock_field_4": "",
+        "expiry_date": "2018-03-19",
+        "goods_received_ID": "3486239A597646B2B7259D91A24988E8",
+        "is_authorised": true,
+        "item_ID": "8F252B5884B74888AAB73A0D42C09E7A", 
+        "item_name": "Salbutamol Inhaler",
+        "kit_data": null,
+        "line_number": 3,
+        "location_ID": "cf5812e0c33911eb9757779d39ae2bdb",
+        "manufacturer_ID": "1FB32324AF8049248D929CFB35F255BA",
+        "order_line_ID": "12e889c0f0d211eb8dddb54df6d741hx",
+        "pack_inners_in_outer": 0,
+        "pack_quan_in_inner": 0,
+        "pack_received": 5,
+        "quantity_received": 50,
+        "remoteCustomerInvoiceLineID": "",
+        "spare_note_has_been_actioned": false,
+        "volume_per_pack": 0,
+        "weight_per_pack": 1
+    }"#,
+);
+
+fn goods_received_line_pull_record() -> TestSyncIncomingRecord {
+    TestSyncIncomingRecord::new_pull_upsert(
+        TABLE_NAME,
+        GOODS_RECEIVED_LINE,
+        GoodsReceivedLineRow {
+            id: "917BA6AEC9984FF09F6DB8599CA426B0".to_string(),
+            goods_received_id: "3486239A597646B2B7259D91A24988E8".to_string(),
+            purchase_order_id: "12e889c0f0d211eb8dddb54df6d741hx".to_string(),
+            received_pack_size: 5.0,
+            number_of_packs_received: 50.0,
+            batch: Some("sal_bat_one".to_string()),
+            weight_per_pack: Some(1.0),
+            expiry_date: Some(NaiveDate::from_ymd_opt(2018, 3, 19).unwrap()),
+            line_number: 3,
+            item_link_id: "8F252B5884B74888AAB73A0D42C09E7A".to_string(),
+            item_name: "Salbutamol Inhaler".to_string(),
+            location_id: Some("cf5812e0c33911eb9757779d39ae2bdb".to_string()),
+            volume_per_pack: Some(0.0), //MOTZ could cause a problem if 0 is meant to be None
```

Comment:
I think we should probably treat zero as none for this case a 0 volume isn't actually valid.

Just add     #[serde(deserialize_with = "zero_f64_as_none")]
here's an example: https://github.com/msupply-foundation/open-msupply/blob/6c83ca348c715d46eac3e4dd33d8953227c5a38a/server/service/src/sync/translations/purchase_order.rs#L171